### DISCUSSION TITLE
Wagmi v2: How to mock the wallet client?

### DISCUSSION THREAD

**Discussion by Max-3-7** [wevm/wagmi#3597] [General]
GM,

With wagmi v1, you could override the wallet client using `MockConnector` but I can't find how to do that with v2?

Here is how we set up the app with v1 to run tests with cypress and anvil:
```
const testAccount = '0x...'
const testChain = <chain>
const forkRpcUrl = '<rpc_url>'

const testWalletClient = createWalletClient<
  HttpTransport,
  typeof testChain,
  `0x${string}` | Account
>({
  account: testAccount,
  chain: testChain,
  transport: http(forkRpcUrl)
})

const mockConnector = new MockConnector({
  chains: [testChain],
  options: {
    chainId: testChain.id,
    walletClient: testWalletClient
  }
})

mockConnector.connect({ chainId: testChain.id })
```

Is it still something we can do with v2?

**Comment by Max-3-7** [comment]
It's possible by implementing `getClient` in the connector: https://github.com/wevm/wagmi/blob/main/packages/core/src/connectors/createConnector.ts#L54-L56

It'd be cool if the `mock` connector would support that by default

**Reply by DidykDima** [reply]
Hey @Max-3-7 do you have a sample of the implementation?

**Comment by tmm** [comment]
https://wagmi.sh/react/api/connectors/mock

**Reply by Max-3-7** [reply]
Yeah I tried this but I don't think there is a way to set a wallet client like the mock connector from v1

**Comment by SergeyVolynkin** [comment]
A working example of how to set up `MockConnector` (we use it to test with Cypress in headless mode)

```ts
import { MockConnector } from 'wagmi/connectors/mock'; // wagmi@1.4.13
// ...
const wagmiConfig = createConfig({
  autoConnect: true,
  connectors: [
    new MockConnector({
      chains: [sepolia],
      options: {
        flags: {
          isAuthorized: true,
        },
        walletClient: createWalletClient({
          account: mnemonicToAccount(testWallet.mnemonic),
          chain: sepolia,
          transport: http(),
        }),
      },
    }),
  ],
  publicClient,
});
```

**Reply by hussmartinez** [reply]
But MockConnector is on V1 right?
What wagmi version are you using? and where do you import MockConnector from?

**Reply by SergeyVolynkin** [reply]
Yes, v1
Added import + version in the original message @hussedev

**Reply by hussmartinez** [reply]
Thanks for the clarification 🙏
But the issue is with V2, but in this way we can implement it based on V1
